quarkus-integration/quarkus/deployment/src/main/java/ai/timefold/solver/quarkus/deployment/TimefoldProcessor.java

```diff
@@ -330,6 +331,21 @@ Some solver configs (%s) don't specify a %s class, yet there are multiple availa
         assertTargetClasses(targetList, DotNames.PLANNING_SOLUTION);
     }
 
+    private void assertNodeSharingDisabled(Map<String, SolverConfig> solverConfigMap) {
+        for (var entry : solverConfigMap.entrySet()) {
+            var solverConfig = entry.getValue();
+            if (solverConfig.getScoreDirectorFactoryConfig() != null &&
+                    Boolean.TRUE
+                            .equals(solverConfig.getScoreDirectorFactoryConfig().getConstraintStreamAutomaticNodeSharing())) {
+                throw new IllegalStateException(
+                        ("SolverConfig %s enabled automatic node sharing via SolverConfig, which is not allowed." +
+                                " Enable automatic node sharing with the property %s instead.")
```

Comment:
You probably wanted to get the string on a single line, but I don't think it's necessary.
In fact, the suggestion typically follows on the second line; in which case, you can use the raw string and can avoid the parens and the concat.

Reply:
/me forgot we had multi-line strings in Java 17